### PR TITLE
Sort outputs in GUI and Excel

### PR DESCRIPTION
**Pull request by zaucker**
- fix sorting of outputs to GUI
- handle translations
- sort order of reports parts (store as array)
- simplify and cleanup frontend report generation
- make Excel export generation consistent

### CODE REVIEW

File: frontend/source/class/agrammon/module/output/Reports.js [hunk 1]
```diff
@@ -307,15 +308,15 @@ qx.Class.define('agrammon.module.output.Reports', {
...
-    },
+        __setModelVariant: function(msg) {
+            var model=msg.getData();
+                if (model == 'LU') {
+                this.__submitButton.show();
+                }
+            else {
+                this.__submitButton.exclude();
+                }
+        },
```
Comment (jnthn) [inline frontend/source/class/agrammon/module/output/Reports.js:319]: Indentation looks a bit dodgy here

File: lib/Agrammon/LanguageParser.pm6 [hunk 2]
```diff
@@ -0,0 +1,15 @@
+use v6;
+
+sub parse-lang-values(Str $value --> Hash) is export {
+    my %opt-lang;
+    for (split("\n", $value)) -> $ol {
+        my ($l, $o) = split(/ \s* '=' \s* /, $ol);
+        if not $o {
+            warn "ol=$ol";
```
Comment (jnthn) [inline lib/Agrammon/LanguageParser.pm6:8]: Maybe make it something like `"Failed to parse language value; ol=$ol"`, so it's easier to understand what the warning is about when it happens in 6 months time. :-)

File: lib/Agrammon/Model/Result.pm6 [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+
 class Agrammon::Model::Result {
     has Str $.name;
     has Str $.type;
-    has  %.selector{Str};
-    has %.data;
+    has %.selector{Str};
+    has @!data-order;
+    has %!data-lookup;
     has Str $._order;
```
Comment (jnthn) [inline lib/Agrammon/Model/Result.pm6:11]: I'm a bit curious why the `_` here

File: lib/Agrammon/UI/Web.pm6 [hunk 4]
```diff
@@ -36,52 +35,41 @@ class Agrammon::UI::Web {
         my (@reports, @graphs);
 
         for $!model.evaluation-order -> $module {
-            my @module-inputs;
             for $module.results -> $result {
+                my %result-hash = $result.as-hash;
+
                 my $type = $result.type;
                 my %report = %(
```
Comment (jnthn) [inline lib/Agrammon/UI/Web.pm6:42]: Missed this before, but if you are assigning pairs into a hash, the `%( ... )` around them is redundant.

File: frontend/source/class/agrammon/module/output/Reports.js (hunk 1 above)
Comment (zaucker) [reply frontend/source/class/agrammon/module/output/Reports.js:319]: Fixed.

File: lib/Agrammon/LanguageParser.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/LanguageParser.pm6:8]: Indeed ... done.

File: lib/Agrammon/Model/Result.pm6 (hunk 3 above)
Comment (zaucker) [reply lib/Agrammon/Model/Result.pm6:11]: For some reason I had named it that way in the old version. As there is no occurance of it in the latest `.nhd` files, I changed it to `order`. We might not even need that anymore, but if we do, we can always use the more sensible name.
I also noticed that in two more `.js` files which also will need more refactoring/fixing soon, but for this PR I just changed the `_order`.

File: lib/Agrammon/UI/Web.pm6 (hunk 4 above)
Comment (zaucker) [reply lib/Agrammon/UI/Web.pm6:42]: Thanks, fixed.

**Review by jnthn — APPROVED** [review]

**Pull request merged**